pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2193,6 +2193,21 @@ private CompletableFuture<Integer> unloadServiceUnit(NamespaceBundle serviceUnit
             if (serviceUnit.includes(topicName)) {
                 // Topic needs to be unloaded
                 log.info("[{}] Unloading topic", topicName);
+                if (topicFuture.isCompletedExceptionally()) {
```

Comment:
@lhotari 
> Did the exception now change to this one?

Yes, after we marked the current unload bundle as having no owner, the `__change_events ` will still try to get topic, so it will get `ServiceUnitNotReadyException` exception.